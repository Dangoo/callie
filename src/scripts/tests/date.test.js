import {
	parseDate,
	formatDate,
	dateInRange,
	getDaysPerMonth,
	getDatesInMonth,
	getMonthsInYear,
	getYears
} from '../date';
import {
	getDateNames
} from '../helper/i18n';

const expectedDate = new Date('2017-02-01');
const monthsMock = ['January', 'February', 'March', 'April', 'Mai', 'June', 'July', 'August', 'September', 'October', 'November', 'December'];

/**
 * Test for parseDate
 */
test('Parse date in format DD.MM.YYYY', () => {
	expect(parseDate('01.02.2017', 'DD.MM.YYYY')).toEqual(expectedDate);
});

test('Parse date in format DD/MM/YYYY', () => {
	expect(parseDate('01/02/2017', 'DD/MM/YYYY')).toEqual(expectedDate);
});

test('Parse date in format MM/DD/YYYY', () => {
	expect(parseDate('02/01/2017', 'MM/DD/YYYY')).toEqual(expectedDate);
});

test('Parse date in format MM.DD.YYYY', () => {
	expect(parseDate('02.01.2017', 'MM.DD.YYYY')).toEqual(expectedDate);
});

/**
 * Test for formatDate
 */
test('Format date in locale "de" / format "DD.MM.YYYY"', () => {
	expect(formatDate(expectedDate, 'de', 'DD.MM.YYYY')).toBe('01.02.2017');
});

test('Format date in locale "en" / format "MM/DD/YYYY"', () => {
	expect(formatDate(expectedDate, 'en', 'MM/DD/YYYY')).toBe('02/01/2017');
});

test('Format date in locale "en-GB" / format "DD/MM/YYYY"', () => {
	expect(formatDate(expectedDate, 'en-GB', 'DD/MM/YYYY')).toBe('01/02/2017');
});

/**
 * Test for dateInRange
 */
test('2017-02-01 in range of 2017-01-01 to 2017-02.02', () => {
	expect(
		dateInRange(
			expectedDate,
			new Date('2017-01-01'),
			new Date('2017-02-02')
		)
	).toBeTruthy();
});

test('2017-02-01 in range of 2017-01-02 to 2017-02.03', () => {
	expect(
		dateInRange(
			expectedDate,
			new Date('2017-02-02'),
			new Date('2017-02-03')
		)
	).toBeFalsy();
});

test(`2017-02-01 without min/max`, () => {
	expect(
		dateInRange(expectedDate)
	).toBeTruthy();
});

/**
 * Test for getDaysPerMonth
 */
test('February 2017 has 28 days', () => {
	expect(getDaysPerMonth(expectedDate)).toBe(28);
});

/**
 * Test for getDatesInMonth
 */
test('Array with 28 entries for Febrary 2017', () => {
	expect(getDatesInMonth(expectedDate)).toMatchSnapshot();
});

/**
 * Test for getMonthsInYear
 */
test('Array with 11 month entries', () => {
	expect(
		getMonthsInYear(monthsMock, expectedDate)
	).toMatchSnapshot();
});

/**
 * Test for getYears
 */
test('Array with 21 year entries', () => {
	expect(
		getYears(2000, 2020, expectedDate.getFullYear())
	).toMatchSnapshot();
});
