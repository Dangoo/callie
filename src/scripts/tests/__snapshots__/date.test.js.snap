// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Array with 11 month entries 1`] = `
Array [
  Object {
    "current": false,
    "disabled": false,
    "month": 0,
    "monthName": "January",
    "selected": false,
  },
  Object {
    "current": false,
    "disabled": false,
    "month": 1,
    "monthName": "February",
    "selected": true,
  },
  Object {
    "current": false,
    "disabled": false,
    "month": 2,
    "monthName": "March",
    "selected": false,
  },
  Object {
    "current": false,
    "disabled": false,
    "month": 3,
    "monthName": "April",
    "selected": false,
  },
  Object {
    "current": true,
    "disabled": false,
    "month": 4,
    "monthName": "Mai",
    "selected": false,
  },
  Object {
    "current": false,
    "disabled": false,
    "month": 5,
    "monthName": "June",
    "selected": false,
  },
  Object {
    "current": false,
    "disabled": false,
    "month": 6,
    "monthName": "July",
    "selected": false,
  },
  Object {
    "current": false,
    "disabled": false,
    "month": 7,
    "monthName": "August",
    "selected": false,
  },
  Object {
    "current": false,
    "disabled": false,
    "month": 8,
    "monthName": "September",
    "selected": false,
  },
  Object {
    "current": false,
    "disabled": false,
    "month": 9,
    "monthName": "October",
    "selected": false,
  },
  Object {
    "current": false,
    "disabled": false,
    "month": 10,
    "monthName": "November",
    "selected": false,
  },
  Object {
    "current": false,
    "disabled": false,
    "month": 11,
    "monthName": "December",
    "selected": false,
  },
]
`;

exports[`Array with 21 year entries 1`] = `
Array [
  Object {
    "current": false,
    "selected": false,
    "year": 2020,
  },
  Object {
    "current": false,
    "selected": false,
    "year": 2019,
  },
  Object {
    "current": false,
    "selected": false,
    "year": 2018,
  },
  Object {
    "current": true,
    "selected": true,
    "year": 2017,
  },
  Object {
    "current": false,
    "selected": false,
    "year": 2016,
  },
  Object {
    "current": false,
    "selected": false,
    "year": 2015,
  },
  Object {
    "current": false,
    "selected": false,
    "year": 2014,
  },
  Object {
    "current": false,
    "selected": false,
    "year": 2013,
  },
  Object {
    "current": false,
    "selected": false,
    "year": 2012,
  },
  Object {
    "current": false,
    "selected": false,
    "year": 2011,
  },
  Object {
    "current": false,
    "selected": false,
    "year": 2010,
  },
  Object {
    "current": false,
    "selected": false,
    "year": 2009,
  },
  Object {
    "current": false,
    "selected": false,
    "year": 2008,
  },
  Object {
    "current": false,
    "selected": false,
    "year": 2007,
  },
  Object {
    "current": false,
    "selected": false,
    "year": 2006,
  },
  Object {
    "current": false,
    "selected": false,
    "year": 2005,
  },
  Object {
    "current": false,
    "selected": false,
    "year": 2004,
  },
  Object {
    "current": false,
    "selected": false,
    "year": 2003,
  },
  Object {
    "current": false,
    "selected": false,
    "year": 2002,
  },
  Object {
    "current": false,
    "selected": false,
    "year": 2001,
  },
  Object {
    "current": false,
    "selected": false,
    "year": 2000,
  },
]
`;

exports[`Array with 28 entries for Febrary 2017 1`] = `
Array [
  Object {
    "current": false,
    "date": 1,
    "day": 3,
    "disabled": false,
    "fullDate": 2017-02-01T00:00:00.000Z,
    "selected": false,
  },
  Object {
    "current": false,
    "date": 2,
    "day": 4,
    "disabled": false,
    "fullDate": 2017-02-02T00:00:00.000Z,
    "selected": false,
  },
  Object {
    "current": false,
    "date": 3,
    "day": 5,
    "disabled": false,
    "fullDate": 2017-02-03T00:00:00.000Z,
    "selected": false,
  },
  Object {
    "current": false,
    "date": 4,
    "day": 6,
    "disabled": false,
    "fullDate": 2017-02-04T00:00:00.000Z,
    "selected": false,
  },
  Object {
    "current": false,
    "date": 5,
    "day": 0,
    "disabled": false,
    "fullDate": 2017-02-05T00:00:00.000Z,
    "selected": false,
  },
  Object {
    "current": false,
    "date": 6,
    "day": 1,
    "disabled": false,
    "fullDate": 2017-02-06T00:00:00.000Z,
    "selected": false,
  },
  Object {
    "current": false,
    "date": 7,
    "day": 2,
    "disabled": false,
    "fullDate": 2017-02-07T00:00:00.000Z,
    "selected": false,
  },
  Object {
    "current": false,
    "date": 8,
    "day": 3,
    "disabled": false,
    "fullDate": 2017-02-08T00:00:00.000Z,
    "selected": false,
  },
  Object {
    "current": false,
    "date": 9,
    "day": 4,
    "disabled": false,
    "fullDate": 2017-02-09T00:00:00.000Z,
    "selected": false,
  },
  Object {
    "current": false,
    "date": 10,
    "day": 5,
    "disabled": false,
    "fullDate": 2017-02-10T00:00:00.000Z,
    "selected": false,
  },
  Object {
    "current": false,
    "date": 11,
    "day": 6,
    "disabled": false,
    "fullDate": 2017-02-11T00:00:00.000Z,
    "selected": false,
  },
  Object {
    "current": false,
    "date": 12,
    "day": 0,
    "disabled": false,
    "fullDate": 2017-02-12T00:00:00.000Z,
    "selected": false,
  },
  Object {
    "current": false,
    "date": 13,
    "day": 1,
    "disabled": false,
    "fullDate": 2017-02-13T00:00:00.000Z,
    "selected": false,
  },
  Object {
    "current": false,
    "date": 14,
    "day": 2,
    "disabled": false,
    "fullDate": 2017-02-14T00:00:00.000Z,
    "selected": false,
  },
  Object {
    "current": false,
    "date": 15,
    "day": 3,
    "disabled": false,
    "fullDate": 2017-02-15T00:00:00.000Z,
    "selected": false,
  },
  Object {
    "current": false,
    "date": 16,
    "day": 4,
    "disabled": false,
    "fullDate": 2017-02-16T00:00:00.000Z,
    "selected": false,
  },
  Object {
    "current": false,
    "date": 17,
    "day": 5,
    "disabled": false,
    "fullDate": 2017-02-17T00:00:00.000Z,
    "selected": false,
  },
  Object {
    "current": false,
    "date": 18,
    "day": 6,
    "disabled": false,
    "fullDate": 2017-02-18T00:00:00.000Z,
    "selected": false,
  },
  Object {
    "current": false,
    "date": 19,
    "day": 0,
    "disabled": false,
    "fullDate": 2017-02-19T00:00:00.000Z,
    "selected": false,
  },
  Object {
    "current": false,
    "date": 20,
    "day": 1,
    "disabled": false,
    "fullDate": 2017-02-20T00:00:00.000Z,
    "selected": false,
  },
  Object {
    "current": false,
    "date": 21,
    "day": 2,
    "disabled": false,
    "fullDate": 2017-02-21T00:00:00.000Z,
    "selected": false,
  },
  Object {
    "current": false,
    "date": 22,
    "day": 3,
    "disabled": false,
    "fullDate": 2017-02-22T00:00:00.000Z,
    "selected": false,
  },
  Object {
    "current": false,
    "date": 23,
    "day": 4,
    "disabled": false,
    "fullDate": 2017-02-23T00:00:00.000Z,
    "selected": false,
  },
  Object {
    "current": false,
    "date": 24,
    "day": 5,
    "disabled": false,
    "fullDate": 2017-02-24T00:00:00.000Z,
    "selected": false,
  },
  Object {
    "current": false,
    "date": 25,
    "day": 6,
    "disabled": false,
    "fullDate": 2017-02-25T00:00:00.000Z,
    "selected": false,
  },
  Object {
    "current": false,
    "date": 26,
    "day": 0,
    "disabled": false,
    "fullDate": 2017-02-26T00:00:00.000Z,
    "selected": false,
  },
  Object {
    "current": false,
    "date": 27,
    "day": 1,
    "disabled": false,
    "fullDate": 2017-02-27T00:00:00.000Z,
    "selected": false,
  },
  Object {
    "current": false,
    "date": 28,
    "day": 2,
    "disabled": false,
    "fullDate": 2017-02-28T00:00:00.000Z,
    "selected": false,
  },
]
`;
